### PR TITLE
Refactor/waitlist

### PR DESCRIPTION
**Pull request by mxkaske**
- installed resend default template
- installed upstash and refactored `server action`
- notification toast

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "pipeline": {
     "build": {
       "dependsOn": ["^build"],
+      "env": ["*", "RESEND_API_KEY", "!NEXT_PUBLIC_GIT_*"],
```
Comment (mxkaske) [inline turbo.json:7]: [Allows `regex`](https://turbo.build/repo/docs/core-concepts/caching/environment-variable-inputs#examples) but somehow, doesn't recognise resend api key.

File: apps/web/src/app/action.ts [hunk 2]
```diff
@@ -1,18 +1,56 @@
...
+export async function addToWaitlist(data: FormData) {
   const email = data.get("email");
   if (email) {
-    await wait(3000);
+    const number = await write(email);
+    await wait(500);
     // TODO: save email to Highstorm
+
+    // REMINDER: how to send emails in server action
+    // send(email)
```
Comment (mxkaske) [inline apps/web/src/app/action.ts:18]: Depends what the plan is. Having it stored in Redis is always good. See [susbscription](https://console.upstash.com/redis/eed2a8c1-b91e-4d5a-9ec9-0ba4ab1b1287?tab=data-browser) database!

File: apps/web/src/app/action.ts [hunk 3]
```diff
@@ -1,18 +1,56 @@
...
+const redis = Redis.fromEnv();
+
+const resend = new Resend(process.env.RESEND_API_KEY);
+
+export async function addToWaitlist(data: FormData) {
   const email = data.get("email");
   if (email) {
-    await wait(3000);
+    const number = await write(email);
+    await wait(500);
```
Comment (thibaultleouay) [inline apps/web/src/app/action.ts:14]: Why do we wait :)

Comment (mxkaske) [reply apps/web/src/app/action.ts:14]: It's too fast otherwise. Creates a flicker

**Review by thibaultleouay — APPROVED** [review]

**Pull request merged**